language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java

```diff
@@ -82,13 +83,17 @@ public static List<CodeAction> getAvailableCodeActions(CodeActionContext ctx) {
                     .setStatementNode(analyzer.getStatementNode().orElse(null))
                     .build();
 
-            codeActionProvidersHolder.getActiveNodeBasedProviders(matchedNodeType, ctx).forEach(provider -> {
+            codeActionProvidersHolder.getActiveRangeBasedProviders(syntaxKind, ctx).forEach(provider -> {
                 try {
                     // Check whether the code action request has been cancelled
                     // in order to avoid unnecessary calculations
                     ctx.checkCancelled();
 
-                    List<CodeAction> codeActionsOut = provider.getNodeBasedCodeActions(ctx, posDetails);
+                    RangeBasedCodeActionProvider rangeBasedProvider = (RangeBasedCodeActionProvider) provider;
```

Comment:
When we keep these as `RangeBasedCodeActionProvider`s list in the holder, no need to cast it here

Reply:
Removed the cast here and casted inside the holder